### ISSUE TITLE
Unable to install wtf8 using npm

### ISSUE THREAD

**Issue by neowulf** [lautis/wtf8#1]
Not able to install wtf8 using npm. Are there any missing dependencies?

```
$ npm install -g wtf8
npm http GET http://registry.npmjs.org/wtf8
npm http 304 http://registry.npmjs.org/wtf8

> wtf8@0.1.1 preinstall /home/user/.nvm/v0.8.0/lib/node_modules/lib/node_modules/wtf8
> node-gyp configure build

make: Entering directory `/home/user/.nvm/v0.8.0/lib/node_modules/lib/node_modules/wtf8/build'
  CXX(target) Release/obj.target/wtf8/wtf8.o
In file included from ../wtf8.cc:1:0:
/home/user/.node-gyp/0.11.7/deps/v8/include/v8.h: In member function 'void v8::ReturnValue<T>::Set(uint32_t)':
/home/user/.node-gyp/0.11.7/deps/v8/include/v8.h:5928:31: warning: typedef 'I' locally defined but not used [-Wunused-local-typedefs]
   typedef internal::Internals I;
                               ^
../wtf8.cc: In function 'v8::Handle<v8::Value> {anonymous}::Decode(const v8::Arguments&)':
../wtf8.cc:37:41: warning: narrowing conversion of '((chr >> 10) | 55296u)' from 'uint32_t {aka unsigned int}' to 'uint16_t {aka short unsigned int}' inside { } is ill-formed in C++11 [-Wnarrowing]
         uint16_t surrogate[2] = {0xD800 | (chr >> 10), 0xDC00 | (chr & 0x3FF)};
                                         ^
../wtf8.cc:37:63: warning: narrowing conversion of '((chr & 1023u) | 56320u)' from 'uint32_t {aka unsigned int}' to 'uint16_t {aka short unsigned int}' inside { } is ill-formed in C++11 [-Wnarrowing]
         uint16_t surrogate[2] = {0xD800 | (chr >> 10), 0xDC00 | (chr & 0x3FF)};
                                                               ^
../wtf8.cc: In function 'v8::Handle<v8::Value> {anonymous}::Encode(const v8::Arguments&)':
../wtf8.cc:105:11: error: expected primary-expression before '&' token
     Buffer& slowBuffer = *Buffer::New(accumulator.size());
           ^
../wtf8.cc:105:13: error: 'slowBuffer' was not declared in this scope
     Buffer& slowBuffer = *Buffer::New(accumulator.size());
             ^
../wtf8.cc: In function 'void {anonymous}::RegisterModule(v8::Handle<v8::Object>)':
../wtf8.cc:113:74: warning: 'static v8::Local<v8::FunctionTemplate> v8::FunctionTemplate::New(v8::InvocationCallback, v8::Handle<v8::Value>, v8::Handle<v8::Signature>, int)' is deprecated (declared at /home/user/.node-gyp/0.11.7/deps/v8/include/v8.h:3354) [-Wdeprecated-declarations]
     target->Set(String::NewSymbol("decode"), FunctionTemplate::New(Decode)->GetFunction());
                                                                          ^
../wtf8.cc:114:74: warning: 'static v8::Local<v8::FunctionTemplate> v8::FunctionTemplate::New(v8::InvocationCallback, v8::Handle<v8::Value>, v8::Handle<v8::Signature>, int)' is deprecated (declared at /home/user/.node-gyp/0.11.7/deps/v8/include/v8.h:3354) [-Wdeprecated-declarations]
     target->Set(String::NewSymbol("encode"), FunctionTemplate::New(Encode)->GetFunction());
                                                                          ^
../wtf8.cc: In function 'v8::Handle<v8::Value> {anonymous}::Encode(const v8::Arguments&)':
../wtf8.cc:110:3: warning: control reaches end of non-void function [-Wreturn-type]
   }
   ^
make: *** [Release/obj.target/wtf8/wtf8.o] Error 1
make: Leaving directory `/home/user/.nvm/v0.8.0/lib/node_modules/lib/node_modules/wtf8/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/home/user/.nvm/v0.11.7/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:267:23)
gyp ERR! stack     at ChildProcess.EventEmitter.emit (events.js:106:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:827:12)
gyp ERR! System Linux 3.11.0-15-generic
gyp ERR! command "node" "/home/user/.nvm/v0.11.7/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "configure" "build"
gyp ERR! cwd /home/user/.nvm/v0.8.0/lib/node_modules/lib/node_modules/wtf8
gyp ERR! node -v v0.11.7
gyp ERR! node-gyp -v v0.10.9
gyp ERR! not ok 
npm ERR! weird error 1
npm ERR! not ok code 0
```

**Comment by eendeego** [comment]
Hey @lautis, would you accept a patch for wtf8 using rvagg/nan for native bindings ? Would solve this issue.

**Comment by lautis** [comment]
I would, although I definitely recommend using built-in Buffer in Node 0.10 or newer. These versions should be able to handle Unicode characters outside basic multilingual plane without any extra libraries.

**Comment by lautis** [comment]
Fixed in 0.2.0.
